### PR TITLE
Early CLI installer

### PR DESCRIPTION
**Pull request by Nicofuma**
capture du 2015-06-03 16 58 33
capture du 2015-06-03 16 59 02

### CODE REVIEW

File: phpBB/install/helper/iohandler/cli_iohandler.php [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function send_response()
+	{
+	}
+
+	public function add_error_message($error_title, $error_description = false)
```
Comment (CHItA) [inline phpBB/install/helper/iohandler/cli_iohandler.php:125]: dockblock

File: phpBB/install/helper/iohandler/cli_iohandler.php [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+	{
+		$this->io->error($this->translate_message($error_title, $error_description));
+	}
+
+	public function add_warning_message($warning_title, $warning_description = false)
```
Comment (CHItA) [inline phpBB/install/helper/iohandler/cli_iohandler.php:142]: dockblock

File: phpBB/install/helper/iohandler/cli_iohandler.php [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	{
+		$this->io->warning($this->translate_message($warning_title, $warning_description));
+	}
+
+	public function add_log_message($log_title, $log_description = false)
```
Comment (CHItA) [inline phpBB/install/helper/iohandler/cli_iohandler.php:159]: dockblock

File: phpBB/install/helper/iohandler/cli_iohandler.php [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+	{
+		$this->io->writeln($this->translate_message($log_title, $log_description));
+	}
+
+
```
Comment (CHItA) [inline phpBB/install/helper/iohandler/cli_iohandler.php:176]: Remove extra new lines

File: phpBB/install/phpbbcli.php [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+$phpbb_installer_container_builder = new \phpbb\di\container_builder($phpbb_root_path, $phpEx);
+$phpbb_installer_container = $phpbb_installer_container_builder
+	->with_environment('installer')
+	->without_extensions()
+	->without_cache()
+	->get_container();
+
+// Path to templates
+$paths = array($phpbb_root_path . 'install/update/new/adm/style', $phpbb_admin_path . 'style');
+$paths = array_filter($paths, 'is_dir');
```
Comment (CHItA) [inline phpBB/install/phpbbcli.php]: the above two lines probably unnecessary?

**Comment by Nicofuma** [comment]
still WIP

File: phpBB/install/helper/config.php [hunk 6]
```diff
@@ -137,6 +137,11 @@ public function system_get($param_name)
 	 */
 	public function get_time_remaining()
 	{
+		if ($this->system_data['max_execution_time'] <= 0)
```
Comment (CHItA) [inline phpBB/install/helper/config.php:140]: Is this supposed to work?

Comment (Nicofuma) [inline phpBB/install/helper/config.php:140]: you can set it to -1 for unlimited

**Comment by CHItA** [comment]
Merged, please feel free to send any improvements via another PR

**Pull request closed without merging**